versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/indexes/StoreIndexImpl.java

```diff
@@ -479,25 +488,28 @@ private ByteBuffer serializeKey(
 
   static <V> StoreIndex<V> deserializeStoreIndex(ByteBuffer serialized, ElementSerializer<V> ser) {
     byte version = serialized.get();
-    checkArgument(version == 1, "Unsupported serialized representation of KeyIndexSegment");
+    checkArgument(
+        version == 1 || version == 2, "Unsupported serialized representation of KeyIndexSegment");
 
     int posPre = serialized.position();
 
+    List<StoreIndexElement<V>> elements =
+        version >= 2 ? new ArrayList<>(readVarInt(serialized)) : new ArrayList<>();
```

Comment:
nit: Could we have `readVarInt` on a separate line as its impact is more on the ser/deser side that on the array's default size.